src/final_chain/final_chain.cpp

```diff
@@ -117,7 +117,8 @@ class FinalChainImpl final : public FinalChain {
         }
       }
     }
-    transaction_count_hint_ = max(transaction_count_hint_, to_execute.size());
+    uint32_t to_execute_trxs_size = to_execute.size();
```

Comment:
why to use this type conversion from size_t to uint32_t when you can just declare transaction_count_hint_ as size_t and keep this part of code as it was ?